src/pybind/mgr/dashboard/services/rbd.py

```diff
@@ -408,34 +409,74 @@ def _rbd_image_stat_removing(cls, ioctx, pool_name, namespace, image_id):
                                 errno=errno.ENOENT)
 
     @classmethod
-    @ViewCache()
-    def rbd_pool_list(cls, pool_name, namespace=None):
+    def _rbd_pool_image_refs(cls, pool_names: List[str], namespace=None):
+        joint_refs = []
         rbd_inst = rbd.RBD()
-        with mgr.rados.open_ioctx(pool_name) as ioctx:
-            result = []
-            if namespace:
-                namespaces = [namespace]
-            else:
-                namespaces = rbd_inst.namespace_list(ioctx)
-                # images without namespace
-                namespaces.append('')
-            for current_namespace in namespaces:
-                ioctx.set_namespace(current_namespace)
-                image_refs = cls._rbd_image_refs(ioctx)
-                for image_ref in image_refs:
+        for pool in pool_names:
+            with mgr.rados.open_ioctx(pool) as ioctx:
+                if namespace:
+                    namespaces = [namespace]
+                else:
+                    namespaces = rbd_inst.namespace_list(ioctx)
+                    # images without namespace
+                    namespaces.append('')
+                for current_namespace in namespaces:
+                    ioctx.set_namespace(current_namespace)
+                    image_refs = cls._rbd_image_refs(ioctx)
+                    for image in image_refs:
+                        image['namespace'] = current_namespace
+                        image['pool'] = pool
+                        joint_refs.append(image)
+        return joint_refs
```

Comment:
I'm curious... If we removed all kind of sorting & filtering, what about simply relying on the order provided by librbd? Would that speed that up? Pool or namespace sorting could be resolved in other ways, since it's a bit of contradictory to iterate all pools and namespaces and then perform a sorting or filtering per pool or namespace. If a user wants that kind of information, we could use [Patternfly filter groups](https://www.patternfly.org/v4/components/toolbar/design-guidelines/#filter-group) with [single-selection filters](https://www.patternfly.org/v4/guidelines/filters/#single-select).

Reply:
What librbd order? If I'm not wrong it doesn't sort anything there. Furthermore, even though this looks anti-performant this function is the most lightweight one (6% of `rbd_pool_list` time)